prospector/filter_rank/ranker.py

```diff
@@ -0,0 +1,94 @@
+from datamodel.advisory import AdvisoryRecord
+from datamodel.commit import Commit
+from filter_rank.utils.model_loader import save_model
+from filter_rank import TRAINING_DATA, NUM_ELEMENTS_TRAINING_DATA
+
+import re
+import pandas as pd
+
+
+def filter_commits(adv_record: AdvisoryRecord, candidates: "list[Commit]") -> "list[Commit]":
+    """
+    Takes in input a set of candidate (datamodel) commits (coming from the commitdb)
+    and returns in output a filtered list obtained by discarding the irrelevant
+    ones based on different criteria (timestamp of commit compared to advisory record date,
+    extensions of the files modified in the commit, and the like)
+    """
+
+    return candidates
+
+
+def rank(adv_record: AdvisoryRecord, candidates: "list[Commit]") -> "list[Commit]":
+    """
+    Takes in input a set of candidates and associates to each of them a rank (ordering) and
+    a ranking vector, based on how good they match with the advisory record in input.
+    """
+    return candidates
+
+
+def train(model_name: str, data_filename=TRAINING_DATA, num_elem_training_data=NUM_ELEMENTS_TRAINING_DATA) -> str:
+    """
+    Takes in input a dataset and produces a trained model that is persisted to disk.
+    Dataset file contains triples:  (vuln_id,fix_id,repository,commit_id)
+    Returns the path to the saved model
+    """
+    # read data_filename content in list of lines
+    # foreach l in line:
+    #   get (commit_id,repo) from l
+    #   if (commit_id,repo) not found in commitdb:
+    #     commit_obj = make_commit(commit_id,repo)
+    #     preprocess(commit_obj)
+    #     save commit_obj to db
+    #   augment commit_obj with advisory-dependent features
+    #   add augmented commit_obj to dataframe
+    # fit(dataframe) --> model
+    # compute metrics
+    # save model to disk (file: model_name)
+    df = _get_training_dataframe(data_filename, num_elem_training_data)
+    if df is None:
+        return ""  # Special case, when returned dataframe is None
+
+    model = ""  # here should go the actual classifier model
+    # model.fit(df)
+    # compute metrics
+    return save_model(model, model_name)
+
+
+def _get_training_dataframe(data_filename=TRAINING_DATA, num_elem_training_data=NUM_ELEMENTS_TRAINING_DATA):
```

Comment:
I would change this to "make_dataframe(...)" because:
- this is not specific to training data
- this does not get (=retrieve/obtain something that exists) but rather constructs it (and in fact, if you need
 the dataframe again, you wouldn't call this function again -- the semantics of make is that once you made it,
 you have it, and you get it every time you need it)

Reply:
Sure. I have just renamed this